app/controllers/pay.ts

```diff
@@ -26,29 +26,33 @@ export default class PayController extends Controller {
     return {
       pricingFrequency: this.selectedPricingFrequency,
       regionalDiscount: this.shouldApplyRegionalDiscount ? this.model.regionalDiscount : null,
-      earlyBirdDiscountEnabled: this.selectedPricingFrequency === 'yearly' && this.user.isEligibleForEarlyBirdDiscount,
-      referralDiscountEnabled: this.selectedPricingFrequency === 'yearly' && this.user.isEligibleForReferralDiscount,
+      earlyBirdDiscountEnabled: this.selectedPricingFrequency === 'yearly' && this.user?.isEligibleForEarlyBirdDiscount,
+      referralDiscountEnabled: this.selectedPricingFrequency === 'yearly' && this.user?.isEligibleForReferralDiscount,
     };
   }
 
   get discountedYearlyPrice() {
-    if (this.user.isEligibleForReferralDiscount) {
+    if (this.user?.isEligibleForReferralDiscount) {
       return 216;
-    } else if (this.user.isEligibleForEarlyBirdDiscount) {
+    } else if (this.user?.isEligibleForEarlyBirdDiscount) {
       return 216;
     } else {
       return null;
     }
   }
 
   get user() {
-    return this.authenticator.currentUser as UserModel; // pay route is protected by auth, so currentUser is guaranteed to be a UserModel
+    return this.authenticator.currentUser as UserModel | null;
```

Comment:
```suggestion
    return this.authenticator.currentUser;
```

(avoid `as` ensures that if the contract changes in the future, we'll start seeing type errors)